### ISSUE TITLE
Adding special tokens to vocabulary

### ISSUE THREAD

**Issue by rafael-wampfler** [axolotl-ai-cloud/axolotl#907]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

The special token `###` should be added to the vocabulary.

### Current behaviour

It is unclear how to add a special token to the vocabulary.

### Steps to reproduce

I'm fine-tuning Llama-2 13b with Axolotl. My dataset for fine-tuning look as follows:

```
# dataset.jsonl 
{"text": "### Human: This is a question### Chatbot: This is a reply### Human: What the hell are you talking about?"} 
{"text": "### Human: Who's coming tonight?### Chatbot: No one, it's literally Monday."} 
... 
```
I'm using `###` as special tokens to separate turns. How can I add `###` to the vocabulary during training with Axolotl? Should I add it to the special_tokens in the yaml config file?

### Config yaml

```
base_model: meta-llama/Llama-2-13b-chat-hf
base_model_config: meta-llama/Llama-2-13b-chat-hf
model_type: LlamaForCausalLM
tokenizer_type: LlamaTokenizer
is_llama_derived_model: true

load_in_8bit: false
load_in_4bit: true
strict: false

datasets:
  - path: myusername/einstein-llama
    type: alpaca
    field: text
dataset_prepared_path: last_run_prepared
hub_model_id: myusername/einstein-llama
val_set_size: 0.01
output_dir: ./qlora-out

adapter: qlora
lora_model_dir:

sequence_len: 4096
sample_packing: true
pad_to_sequence_len: true

lora_r: 32
lora_alpha: 16
lora_dropout: 0.05
lora_target_modules:
lora_target_linear: true
lora_fan_in_fan_out:

wandb_project: "llama-einstein"
wandb_entity:
wandb_watch:
wandb_run_id:
wandb_log_model: "checkpoint"

gradient_accumulation_steps: 4
micro_batch_size: 2
num_epochs: 3
optimizer: paged_adamw_32bit
lr_scheduler: cosine
learning_rate: 0.0002

train_on_inputs: false
group_by_length: false
bf16: true
fp16: false
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_steps: 10
eval_steps: 0.05
eval_table_size:
save_steps:
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
  bos_token: "<s>"
  eos_token: "</s>"
  unk_token: "<unk>"
```

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main/575a082

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
You can add it under 
```yml
tokens:
  - "###"
```

although not really necessary.

**Comment by generalsvr** [comment]
> special_tokens:

What is the difference between `special_tokens` and `tokens`?

**Comment by NanoCode012** [comment]
special tokens are the bos/eos/pad/unk tokens
